### PR TITLE
fix(schemas): modify domain table unique index

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Per discussion, modify domain table unique index to `(tenant_id, domain)`

## Testing

Local tested

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

**Review by simeng-li — COMMENTED** [review]
@wangsijie need to rebase and re-order the alteration time stamp.

File: packages/schemas/alterations/next-1685691718-domain-unique.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      alter table domains drop constraint domains__domain;
+      create unique index domains__domain ON domains using btree (tenant_id,domain);
```
Comment (gao-sun) [inline packages/schemas/alterations/next-1685691718-domain-unique.ts:9]:
```suggestion
      create unique index domains__domain on domains (tenant_id, domain);
```

File: packages/schemas/alterations/next-1685691718-domain-unique.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    `);
+  },
+  down: async (pool) => {
+    await pool.query(sql`
+      alter table domains drop constraint domains__domain;
+      create unique index domains__domain ON domains using btree (domain);
```
Comment (gao-sun) [inline packages/schemas/alterations/next-1685691718-domain-unique.ts:15]: drop index and create constrant?

Comment (wangsijie) [reply packages/schemas/alterations/next-1685691718-domain-unique.ts]: It is unable to drop "index" if the table has data, drop constraint works

Comment (wangsijie) [reply packages/schemas/alterations/next-1685691718-domain-unique.ts]: image

Comment (gao-sun) [reply packages/schemas/alterations/next-1685691718-domain-unique.ts]: i was blinding by the light

**Pull request merged**